### DISCUSSION TITLE
Need help to use the library in ASP code

### DISCUSSION THREAD

**Discussion by lhac5vet** [schoero/SwissQRBill#383] [General]
Hi Roger,
I'm looking for the right syntax to add the SwissQRBill library in my ASP application (old ERP with a mix of VBScript and javascript). I have added the pre-bundled library  `<script type="text/javascript" src="https://cdn.jsdelivr.net/npm/swissqrbill/lib/browser/bundle/index.js" /> ` but then I'm not able to use it. The `import { PDF } from "swissqrbill/pdf";` doesn't work and I don't know which other syntax I should use. Is it possible to use it in ASP and what should be the correct syntax ?
Thanks,
Laurent
[swissbill.asp.txt](https://github.com/schoero/SwissQRBill/files/9289650/swissbill.asp.txt)

**Comment by schoero** [comment]
Hi Laurent

Im not familiar with ASP, but I assume it should be able to import it normally.

I have just realised that the snipped you tried to use from the readme is not valid HTML.
```diff
-<script type="text/javascript" src="https://cdn.jsdelivr.net/npm/swissqrbill/lib/browser/bundle/index.js"  />
+<script type="text/javascript" src="https://cdn.jsdelivr.net/npm/swissqrbill/lib/browser/bundle/index.js" ></script>
```

You don't need any additional import statement, because you are importing the library with this script tag. After that, there is a global variable named `SwissQRBill` available. You can access the different parts of the library using `SwissQRBill.PDF` for example.

Because there is no file system in the browser, we need to write the output into a buffer. You can use the built in `BlobStream` for that.

You can then convert the buffer into a url using `buffer.toBlobURL("application/pdf");`. You can then load this url in the current window or an iframe.

Here you have a minimal working example:

```html
<head>

  <script type="text/javascript" src="https://cdn.jsdelivr.net/npm/swissqrbill/lib/browser/bundle/index.js" ></script>
  <script type="text/javascript">

    const data = {
      currency: "CHF",
      amount: 1199.95,
      reference: "210000000003139471430009017",
      creditor: {
        name: "Robert Schneider AG",
        address: "Rue du Lac",
        buildingNumber: "1268",
        zip: 2501,
        city: "Biel",
        account: "CH4431999123000889012",
        country: "CH"
      },
      debtor: {
        name: "Pia-Maria Rutschmann-Schnyder",
        address: "Grosse Marktgasse",
        buildingNumber: "28",
        zip: 9400,
        city: "Rorschach",
        country: "CH"
      }
    };

    const stream = new SwissQRBill.BlobStream();
    const pdf = new SwissQRBill.PDF(data, stream);
    
    pdf.on("finish", () => {
      const iframe = document.getElementById("iframe");
      const url = stream.toBlobURL("application/pdf");
      // To load in the iframe
      iframe.src = url;
      // To load in the current window
      // window.location.href = url
    });
    
  </script>
</head>

<body>
  <iframe id="iframe"></iframe>
</body>
```

Let me know if you need any further assistance.

**Reply by lhac5vet** [reply]
Hi Roger,

I have tried a lot of syntaxes but this one is obvious now that I know it. And I can see in your example that I forgot to give an ID to the iframe in my code. It doesn't help too. 
It works like a charm ! 

Thanks a lot for your help

Grüsse aus dem Kanton Waadt
Laurent
